tnc/data_handler.py

```diff
@@ -876,39 +876,8 @@ def arq_transmit(self, data_out: bytes, mode: int, n_frames_per_burst: int):
         while not self.data_frame_ack_received and static.ARQ_STATE:
             # we have TX_N_MAX_RETRIES_PER_BURST attempts for sending a burst
             for self.tx_n_retry_of_burst in range(TX_N_MAX_RETRIES_PER_BURST):
-                # AUTO MODE SELECTION
-                # mode 255 == AUTO MODE
-                # force usage of selected mode
-                if mode != 255:
-                    data_mode = mode
-                    self.log.debug("[TNC] FIXED MODE:", mode=data_mode)
-                else:
-                    # we are doing a modulo check of transmission retries of the actual burst
-                    # every 2nd retry which fails, decreases speedlevel by 1.
-                    # as soon as we received an ACK for the current burst, speed_level will increase
-                    # by 1.
-                    # The intent is to optimize speed by adapting to the current RF conditions.
-                    # if not self.tx_n_retry_of_burst % 2 and self.tx_n_retry_of_burst > 0:
-                    #    self.speed_level = max(self.speed_level - 1, 0)
-
-                    # if self.tx_n_retry_of_burst <= 1:
-                    #    self.speed_level += 1
-                    # self.speed_level = max(self.speed_level + 1, len(self.mode_list) - 1)
-
-                    # Bound speed level to:
-                    # - minimum of either the speed or the length of mode list - 1
-                    # - maximum of either the speed or zero
-                    self.speed_level = min(self.speed_level, len(self.mode_list) - 1)
```

Comment:
This and the lines following shouldn't be removed, I think.

For min and max functions I used np.clip, so I suggest we are using the same functions application wide. There has also be a bug with min/max in another area of the code. So we should test this, I think. I can do this as well the next days as I want to see the possible differences between np.clip and min/max 🤔

Reply:
Restored that section. 

While I don't necessarily object to using the same function application-wide, `np.clip` is really meant for NumPY arrays, not single arrays. I can make the change as I'm sure it will work, but if we need to provide `np.clip` a list and retrieve element 0 from the returned list I would suggest we use min/max instead.

Reply:
Ah yes, you're right. We used np.clip when processing audio data I think. I just want to avoid a possible bug. So it's good if we are talking about this 👍 
I,ll resolve this conversation.